### PR TITLE
feat: Updated to latest cli 2.5.0

### PR DESCRIPTION
**Pull request by Aerex**
# Context
The PR will update the extension to use the latest version of the CLI. In addition, all references to **bluemix** has been rebranded to **IBM Cloud**

# Steps to Test
### Setup
1. Install NPM
2. Install the latest IBM Cloud CLI
3. Install Visual Studio Code Extension Manager CLI (`vsce`)
```
npm install -g vsce
```
4. Install VSCode
```
vsce package
```
5. Change directory to the root of the ibm-developer-extension-vscode project
6. Build extension
7. Open VSCode
8. Click on **Extension** then click on the `...` on the top left side of the panel
9. Click on the Install From VSX in the dropdown
10. Select the `.vsx` file that was built in step 4
11. Verify that the ibm-developer-0.0.15.vsix was created
12. Create a namespace in IBMCloud Container Registry in production
  12.1 You will need this when deploying container
13. Create or use an existing K8 cluster

### Verify rebranding
1. Check sourcecode and README **DOES NOT** reference **bluemix**
  1.1 You should see **bluemix** renamed  to **IBMCloud**

### Test create, running and deploying app development through IKS workflow
1. Open VSCode to a blank page
2. Open up the command palette using the shortcut: *Ctrl+Shift+P*
3. Enter `ibmcloud login --sso` and select command
  3.1 Make sure that you are using production API endpoint
4. Enter `ibmcloud dev create` and select command
5. Verify that terminal window is opened in VSCode
6. Follow and respond to the prompts presented
7. Verify that new project is created.
8. Open up new project in VSCode
9. Open command palette and enter `ibmcloud dev build (release)` and select command
10. Verify that project is built successfully
11. Open command palette and enter `ibmcloud dev run` and select command
12. Verify that you view app by visiting url in browser
13. Set the following properties in the `cli-config,.yml` file
  ```
    deploy-target: "container"
    deploy-image-target: "icr.io/<namespace>/<image-name>"
    ibm-cluster: "<cluster-name>"

  ```
14. Verify that hosted app url is shown in the Output panel
15. Visit hosted app in your browser

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,60 @@
...
-- Use the `bx dev run` command to run the app in local Docker in release mode
-- Use the `bx dev deploy` command to deploy the app (in the Docker container) to Bluemix
-- Use the `bx dev console` command to open your project on Bluemix in a web browser
-
-### Using CloudFoundry workflow: 
+- Use the `ibmcloud dev build` command to build the app into a Docker image
+- Use the `ibmcloud dev debug` command to run the app in local Docker for development
+- Use the `ibmcloud dev run` command to run the app in local Docker in release mode
+- Use the `ibmcloud dev deploy` command to deploy the app (in the Docker container) to IBM Cloud
+- Use the `ibmcloud dev view` command to open your project on IBM Cloud in a web browser
```
Comment (steveclay) [inline README.md:54]: A better description would be `to open your deployed app`

Comment (Aerex) [reply README.md:54]: Will do

File: README.md [hunk 2]
```diff
@@ -60,32 +63,31 @@ Steps to get started:
...
 
-- `bx login/logout` - Bluemix user authentication
-- `bx dev` - Bluemix developer CLI
-- `bx cf` - Bluemix CloudFoundry CLI
-- `bx cs` - Bluemix Container Service CLI
-- `bx sdk` - Bluemix SDK Generation CLI
+- `ibmcloud login/logout` - IBMCloud user authentication
+- `ibmcloud dev` - IBMCloud developer CLI
+- `ibmcloud cf` - IBMCloud CloudFoundry CLI
+- `ibmcloud cs` - IBMCloud Container Service CLI
```
Comment (steveclay) [inline README.md:77]: This is now `ibmcloud ks` - IBM Cloud Kubernetes Service CLI

File: README.md [hunk 3]
```diff
@@ -60,32 +63,31 @@ Steps to get started:
...
 
 ## Requirements/Dependencies
 
-* [Bluemix CLI](https://plugins.ng.bluemix.net/ui/home.html)
-* Bluemix `dev` cli plugin   
-    After installing the Bluemix CLI, open up a terminal and run `bx plugin install dev -r Bluemix`
-* [Docker](https://www.docker.com/) - required by `bx dev` containers
+* [IBMCloud CLI](https://cloud.ibm.com/docs/cli/index.html)
+* IBMCloud `dev` cli plugin
```
Comment (steveclay) [inline README.md:82]: Remove this section about installing **dev** as it is now bundled with the CLI.

File: README.md (hunk 2 above)
Comment (Aerex) [reply README.md:77]: Will do

File: src/extension.ts [hunk 4]
```diff
@@ -36,60 +36,62 @@ let unsupportedDevVersion = false;
...
+
+    LoginManager.registerCommand(context, 'extension.ibmcloud.login');
+    LoginManager.registerCommand(context, 'extension.ibmcloud.login.sso');
+    registerCommand(context, 'extension.ibmcloud.logout', {cmd: 'ibmcloud', args: ['logout']}, outputChannel);
+    registerCommand(context, 'extension.ibmcloud.cli-update', {cmd: 'ibmcloud', args: ['plugin', 'update', '-r', 'IBMCloud']}, outputChannel);
+
+
+    // IBMCloud DEV commands *************************************
+    registerCommand(context, 'extension.ibmcloud.dev.list', {cmd: 'ibmcloud', args: ['dev', 'list', '--caller-vscode']}, outputChannel, true);
+    registerCommand(context, 'extension.ibmcloud.dev.create', {cmd: 'ibmcloud', args: ['dev', 'create', '--caller-vscode']}, outputChannel, true, SystemCommand, true);
```
Comment (steveclay) [inline src/extension.ts:49]: Prior version did not have `create`, but instructed the user to do this step outside of the extension.

File: src/extension.ts [hunk 5]
```diff
@@ -147,27 +149,27 @@ function checkVersions(): Promise<any> {
...
 
-                    // parse version from bx --version command output
+                    // parse version from ibmcloud --version command output
                     const split = command.stdout.split('+');
                     const detail = split[0].split('version');
                     const version = semver.clean(detail[detail.length - 1]);
 
                     if (semver.gt(packageJson.ibm.cli.version, version)) {
-                        const message = `\n\nThe recommended minimum Bluemix CLI version is ${packageJson.ibm.cli.version}.\nYour system is currently running ${version}.\nA newer version of the IBM Bluemix CLI is available for download at: ${packageJson.ibm.cli.url}`;
+                        const message = `\n\nThe recommended minimum IBMCloud CLI version is ${packageJson.ibm.cli.version}.\nYour system is currently running ${version}.\nA newer version of the IBMCloud CLI is available for download at: ${packageJson.ibm.cli.url}`;
```
Comment (steveclay) [inline src/extension.ts:163]: Same comment for these messages -- `IBMCloud` should be `IBM Cloud`

File: src/util/DeployCommand.ts [hunk 6]
```diff
@@ -50,7 +50,7 @@ export class DeployCommand extends SystemCommand {
                     this.displayError('Please specify \'deploy-image-target\' in cli-config.yml for Kubernetes deployment.');
                 });
             } else {
-                const regex = new RegExp(/registry.*.bluemix.net/);
+                const regex = new RegExp(/registry.*.icr.io/);
```
Comment (steveclay) [inline src/util/DeployCommand.ts:53]: I don't believe the registry begins with `registry` anymore. But just <region>.icr.io.

Comment (Aerex) [reply src/util/DeployCommand.ts:53]: I see. I found some inconsistency in this doc [page](https://cloud.ibm.com/docs/cli?topic=cli-idt-cli#deploy) in production.  I can file an issue  with the doc team

File: src/extension.ts (hunk 4 above)
Comment (Aerex) [reply src/extension.ts:49]: Right. Let me remove the new command here. I think we decided to hold off on adding new commands for the new release.

File: README.md (hunk 2 above)
Comment (steveclay) [reply README.md]: Please change to `IBM Cloud Kubernetes Service CLI`

File: src/util/DeployCommand.ts (hunk 6 above)
Comment (steveclay) [reply src/util/DeployCommand.ts]: Yes, please do

File: README.md (hunk 2 above)
Comment (Aerex) [reply README.md]: Sorry I must have missed this here. Will fix.

**Comment by Aerex** [comment]
Ready for re-review @steveclay. I am going to do another round of testing too

File: .travis.yml [hunk 7]
```diff
@@ -19,3 +19,12 @@ before_install:
 
 script:
   - npm test --silent
+
```
Comment (Aerex) [inline .travis.yml:22]: Callout: Added publish script to publish ext only on tagged release.

File: .travis.yml [hunk 8]
```diff
@@ -19,3 +19,12 @@ before_install:
...
+after_script:
+- |
+  echo ">>> Publish"
+  npm run vscode:publish
+
+stages:
+- name: after_script
+  if: env(TRAVIS_TAG) =~ ^v
```
Comment (Aerex) [inline .travis.yml:30]: Callout: Added `after_script` to publish new version of extension on release builds (e. v0.0.15)

**Review by steveclay — APPROVED** [review]

**Pull request merged**